src/crewai/memory/memory.py

```diff
@@ -12,12 +12,16 @@ def __init__(self, storage: Storage):
         self.storage = storage
 
     def save(
-        self, value: Any, metadata: Dict[str, Any] = None, agent: str = None
+        self,
+        value: Any,
+        metadata: Optional[Dict[str, Any]] = None,
+        agent: Optional[str] = None,
```

Comment:
What is your take on `Optional[str]` vs `str | None`? I've been using the latter these days.

Reply:
Yeah. If `| None` is the updated way to do it, let's try to use it. But there's no reason to start switching everywhere. We can adjust as we move forward.